core/src/main/java/org/apache/accumulo/core/spi/balancer/data/TabletStatistics.java

```diff
@@ -28,6 +28,7 @@ public interface TabletStatistics extends Comparable<TabletStatistics> {
 
   long getNumEntries();
 
+  @Deprecated(since = "3.1", forRemoval = true)
```

Comment:
I'm still torn on whether we should bother marking things as "forRemoval" or not, at all. On the one hand, it's nice to be explicit, but on the other hand, all our deprecations are always subject to removal, causing a breaking change on a major release, in accordance with semver--so, it's kind of redundant, and kind of feels like we're just using it because it's there, when we don't really need to.

The other issue is this Eclipse bug that makes it hard to properly suppress "removal". See https://bugs.eclipse.org/bugs/show_bug.cgi?id=565271
Other IDEs may also be affected by improperly handling this new annotation attribute.

Reply:
I created #4032 for this.